src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -2085,22 +2084,13 @@ namespace SurfaceGeometry {
         errFlag = false;
         if (!SurfError) {
             for (int SurfNum = 1; SurfNum <= MovedSurfs; ++SurfNum) { // TotSurfaces
-                // Set ShadedConstruction numbers for windows whose shaded constructions were created
-                // when shading device was specified in the WindowShadingControl for the window
-                if (Surface(SurfNum).ShadedConstruction != 0) SurfWinShadedConstruction(SurfNum) = Surface(SurfNum).ShadedConstruction;
-
-                // no need to set the below -- it is the default
-                // Set variable that indicates if shading device has movable slats
-                //      SurfaceWindow(SurfNum)%MovableSlats = .FALSE.
-
-                // TH 2/9/2010. Fixed for CR 8010 for speed up purpose rather than fixing the problem
-                WinShadingControlPtr = Surface(SurfNum).WindowShadingControlPtr;
                 if (Surface(SurfNum).HasShadeControl) {
+                    WinShadingControlPtr = Surface(SurfNum).activeWindowShadingControl; // use first item since others should be identical
```

Comment:
I think this section needs to cycle through the `windowShadingControlList` in case one of them uses a different type of blind.

Reply:
I'm not sure, the ShadingType and SlatAngleControlForBlinds are two of the items that need to be the same currently in the isWIndowShadingControlSimilar() function. There is only one SurfWinBlindNumber so this may have to stay as the same across WIndowShadingControls.

Reply:
OK. I suppose `SurfWinBlindNumber` could be changed to `SurfWinActiveBlindNumber` and we could add a `blindNumberList` to `Surface`. But this can wait for a Phase II where we can fully test relaxing all of these matching requirements.